app/assets/javascripts/recipes.js

```diff
@@ -11,15 +9,22 @@ $(document).ready(function(){
     });
     if(index > 0 && inputsAreBlank){
       $(grainElement).hide();
-    } else if (index > 0) {
-      $(grainElement).prev(".grain").find("button.add-grain").hide();
-      // Only show button for last non-blank input
     }
-   });
+  });
+
+  var show_last_add_grain_button = function(){
+    $('.grain').filter( function(){ 
+      if($(this).css('display') == "block") {
+        return $(this);
+      }
+    }).last().find('button.add-grain').show();
+  };
+  show_last_add_grain_button();
 
   $("button.add-grain").on("click", function(event) {
     event.preventDefault();
-    $(this).parent(".grain").next(".grain").show();
+    $(this).parent(".grain").nextAll(".grain").first().show();
```

Comment:
Consider only having one `<button class="add-grain">`, as it may reduce the amount of code needed here for managing button visibility. Admittedly it may not too.